### PR TITLE
FTX API endpoint is cinfigurable 

### PR DESCRIPTION
**Pull request by AdalyatNazirov**
#### Description

specify enpoint url of ftx api (can be FTX pr FTX.US). FTX and FTX.US are considered as entirely different exchanges

#### Related Issue

relates #5816 

#### Motivation and Context

FTX is running different exchange (and endpoing) for trading in USA

#### How Has This Been Tested?

N/A

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by AdalyatNazirov** [comment]
> 🤔 I wonder if there's a different way to do this. It won't scale well to have countries carved out of the Market enum. I think we only have 255 slots. Are the FTX US BTCUSD fungible and able to be moved and traded on the FTX exchange?

initially it was the idea, but then we found that FTX and FTX.US are different - not only endpoints, but also different history, listed symbols etc. It made us to change a decision on implementation and create another brokerage - `FTXUSBrokerage: FTXBrokerage`, with overridden bits responsible for diffs. So it will even will have it's own `live-ftxus` section in config file.

File: Common/Brokerages/FTXUSBrokerageModel.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        /// <summary>
+        /// Market name
+        /// </summary>
+        protected override string MarketName => Market.FTXUS;
+
+        /// <summary>
+        /// Gets a map of the default markets to be used for each security type
+        /// </summary>
+        public override IReadOnlyDictionary<SecurityType, string> DefaultMarkets { get; } = GetDefaultMarkets(Market.FTXUS);
```
Comment (Martin-Molinero) [inline Common/Brokerages/FTXUSBrokerageModel.cs:40]: Cool! I think if underlying directly used `MarketName` we wouldn't need to override it again?
nit - mind moving constructor after properties/fields

File: Launcher/config.json [hunk 2]
```diff
@@ -443,6 +450,21 @@
...
       "history-provider": "BrokerageHistoryProvider"
+    },
+
+    // defines the 'live-ftxus' environment
+    "live-ftxus": {
+      "live-mode": true,
+
+      // real brokerage implementations require the BrokerageTransactionHandler
+      "live-mode-brokerage": "QuantConnect.FTXBrokerage.FTXUSBrokerage",
+      "data-queue-handler": "QuantConnect.FTXBrokerage.FTXUSBrokerage",
```
Comment (Martin-Molinero) [inline Launcher/config.json:461]: > "data-queue-handler"

Value should be an array moving forward 👍

File: Launcher/config.json [hunk 3]
```diff
@@ -194,6 +194,11 @@
   "ftx-api-secret": "",
   "ftx-api-key": "",
 
+  // FTX.US configuration
+  "ftxus-account-tier": "Tier1", // accept "Tier1", "Tier2", "Tier3", "Tier4", "Tier5", "Tier6", "VIP1", "VIP2", "VIP3", "MM1", "MM2", "MM3"
```
Comment (Martin-Molinero) [inline Launcher/config.json:198]: They have the same tier values? cool 👍

File: Tests/Common/Brokerages/FTXUSBrokerageModelTests.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+*/
+
+using NUnit.Framework;
+using QuantConnect.Brokerages;
+using QuantConnect.Orders.Fees;
+
+namespace QuantConnect.Tests.Common.Brokerages
+{
+    [TestFixture]
+    public class FTXUSBrokerageModelTests : FTXBrokerageModelTests
```
Comment (Martin-Molinero) [inline Tests/Common/Brokerages/FTXUSBrokerageModelTests.cs:23]: Nice!

**Review by Martin-Molinero — APPROVED** [review]
Sweet! leaving couple of minor comments 👍

File: Common/Brokerages/FTXUSBrokerageModel.cs (hunk 1 above)
Comment (AdalyatNazirov) [reply Common/Brokerages/FTXUSBrokerageModel.cs:35]: it requires `MarkerName` to be static field.

File: Launcher/config.json (hunk 3 above)
Comment (AdalyatNazirov) [reply Launcher/config.json]: Apparently it has more tiers (+7, 8, 9) and only VIP1 and VIP2. Fixed

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**